pkg/registry/node/etcd/etcd.go

```diff
@@ -82,7 +90,11 @@ func NewREST(s storage.Interface, storageDecorator generic.StorageDecorator, con
 	statusStore := *store
 	statusStore.UpdateStrategy = node.StatusStrategy
 
-	return &REST{store, connection, proxyTransport}, &StatusREST{store: &statusStore}
+	return NodeStorage{
+		Node:   &REST{store, connection, proxyTransport},
+		Status: &StatusREST{store: &statusStore},
+		Proxy:  &noderest.ProxyREST{Store: store, Connection: connection, ProxyTransport: proxyTransport},
```

Comment:
set up `nodeREST := &REST{store, connection, proxyTransport}` first. It also satisfies the `client.ConnectionInfoGetter`, but does an additional lookup of the node to determine the kubelet port in use by the node, so it should be passed to `&noderest.ProxyREST{Store: store, Connection: nodeREST, ProxyTransport: proxyTransport}`